### PR TITLE
Store Jenkins configurations

### LINKED ISSUE #2: Store jenkins configuration into this repo
We need to store all our Jenkins configuration into this repo so that we don't loose all our precious configs.

### PR DESCRIPTION
**Pull request by sboeuf**
This PR starts this repository by mainly adding the configurations which need to be stored from the Jenkins server.

Fixes #2

### CODE REVIEW

**Comment by sboeuf** [comment]
@grahamwhaley @jodh-intel @sameo please take a look.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Ignore everything
```
Comment (grahamwhaley) [inline .gitignore:1]: Silly question - why do we want to ignore everything? If we are 'managing' the files through git, then I would have thought we'd like to be able to do a `git diff` and `git status` to see what has been modified, and thus needs committing - no?

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,9 @@
 
 # Don't ignore the LICENSE file
 !LICENSE
+
```
Comment (grahamwhaley) [inline .gitignore]: Ah, OK - wrt my previous comment then - why not merge these two patches - no point excluding everything and then immediately adding files back in.
And this commit doesn't just modify the .gitignore, it adds a bunch of .xml files as well (not noted in the commit message)

File: .gitignore [hunk 3]
```diff
@@ -7,6 +7,9 @@
 # Don't ignore the LICENSE file
 !LICENSE
 
+# Don't ignore the README.md file
```
Comment (grahamwhaley) [inline .gitignore]: Given we are adding things to the .gitignore - I presume there are files that we really do want to ignore? Just wondering if we should just not ignore everything, or if we might be better approaching from the other direction - add ingnores for the things we really do want to ignore?

File: credentials.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          <id>6af1abaa-97d7-4ecf-9616-2d81faa76ac3</id>
+          <description>ubuntu_16_04_azure_slave</description>
+          <username>jenkins</username>
+          <privateKeySource class="com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey$UsersPrivateKeySource"/>
+        </com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey>
+        <org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl plugin="plain-credentials@1.4">
+          <scope>GLOBAL</scope>
+          <id>0fe2a31d-e947-47c3-aa08-6d5e451b61e5</id>
+          <description>jenkins-ci</description>
+          <secret>{AQAAABAAAAAw1wy6VRwbpPj52Ezeucpa8MEU+UjYHGjIYWKb5MvIc5kTeyj4EYLlhyP1ORTk50xD0GEJtRykMn0mpn5jzb+nhw==}</secret>
```
Comment (grahamwhaley) [inline credentials.xml:20]: Just carrying over a comment from the original PR https://github.com/clearcontainers/tests/pull/545
Just want to check we are sure we are not leaking any secrets - I believe we noted that the real secrets are stored in a separate secrets dir under Jenkins, and hence we are OK. Can we just confirm (as there are a lot of files here to check etc.).

**Review by grahamwhaley — CHANGES_REQUESTED** [review]
Just querying if we can squash some of those commits, and if we are approaching the .gitignore from the right direction.
Overall, looking good though.

File: credentials.xml (hunk 4 above)
Comment (sboeuf) [reply credentials.xml:20]: Yes, the secret that you can see here is encrypted with the keys stored inside the `secrets` folder, that's why it's okay to have the credentials.xml file here.

Comment (grahamwhaley) [reply credentials.xml:20]: thx - then lgtm - let me approve

**Review by grahamwhaley — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (sboeuf) [reply .gitignore:1]: Well I have chosen this approach given the fact that our git repo on the Jenkins server is gonna be complicated to handle if we go the other way. If we ignore all the files on purpose, the .gitignore is gonna be more complex, but honestly I am not sure what's the best approach here.
@jodh-intel @sameo any thought ?

**Comment by jodh-intel** [comment]
lgtm.

I came across this earlier which might be of use as it seems to provide jenkins integration for storing config in git:

- https://wiki.jenkins.io/display/JENKINS/SCM+Sync+configuration+plugin

**Comment by grahamwhaley** [comment]
That link to the auto-config save/update plugin looks worth investigating @jodh-intel

**Comment by sboeuf** [comment]
@jodh-intel thanks for the link, I agree this link worthes some investigations ;)

**Pull request merged**